src/v/cloud_storage_clients/util.cc

```diff
@@ -15,8 +15,31 @@
 
 #include <boost/property_tree/xml_parser.hpp>
 
+#include <regex>
```

Comment:
i think this include is a leftover

Reply:
yes, I tried originally to use regex. strangely clangd doesn't complain of unused import for this. removed now.